### ISSUE TITLE
Multi-tenancy - provide way for operators to share hunt without joining

### ISSUE THREAD

**Issue by jpd236** [deathandmayhem/jolly-roger#2281]
I set up a few different "hunts" on our JR instance for different teams working on Galactic Puzzle Hunt over the last week.

What I had to do was:

- Create the hunt
- Use my operator status to join the hunt
- Create an invitation link
- Share the invitation link
- Wait for someone to join
- Make them an operator and tell them to manage their own operators going forward
- Edit MongoDB manually to remove myself from the hunt. Otherwise, I see their announcements and have their puzzle files shared with me, which isn't desirable as I'm not actually solving with them

It would be nice to have a more streamlined flow to create a hunt for somebody else to manage, without requiring that I be part of the hunt myself. (At a minimum, a "Leave hunt" button would have been helpful to avoid the manual DB edit).

FWIW, it's also unfortunate that Discord doesn't provide a way to "leave" a channel, so I have to just mute the other team channels and make sure I avoid going in there. But I don't think there's much JR can do about this.

**Comment by zarvox** [comment]
It sounds like you're primarily involved here because hunt creation is currently restricted to server admins; is that right?  We've had some ideas around allowing hunt creation by anyone by moving some config options from per-server to per-hunt like in https://github.com/deathandmayhem/jolly-roger/issues/630 but that is admittedly a somewhat sizable endeavor.

Would most of these problems go away if e.g. the other operators could create their own hunts (which they'd default be operator of) and make their own sharing links rather than relying on you to provision them for them?  Leaving a hunt feels somewhat complicated because then everything that has a record of a user id needs to deal with that no longer having e.g. a retrievable display name, and I don't think we've really thought through the implications of the creator of a hunt abandoning membership in it before.

**Comment by jpd236** [comment]
More or less, I think that would solve the problem, yeah.

One slight drawback would be if that would require making anyone who wants to add a hunt a full server admin / grants them more privileges than just the ability to create new hunts. But if we were just granting the ability to add new hunts without any other more dangerous operations, that seems fine. Only risk could be a bit more cruft/inconsistency in the hunt list with a larger set of people adding hunts, but that could be manageable in other ways (e.g. searchable hunt list, ability to archive hunts)

**Comment by jimsug** [comment]
Dumb question but would having a dedicated "admin" user be a suitable workaround at the moment? This user is not your usual hunt account, so you don't see puzzles, sheets, etc.

**Comment by jpd236** [comment]
Yeah, I think that works in the interim. Folks will see an "Admin" on their hunt rosters but that's okay. I can also set up a separate Discord account to be server admin so I don't see the team-specific channels.
